src/Twig/Transformer/TwigReplaceTransformer.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+namespace DrupalRector\Twig\Transformer;
+
+use Rector\Core\Provider\CurrentFileProvider;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+final class TwigReplaceTransformer implements TwigTransformer
+{
+
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition('Convert replace function into array', [
+            new CodeSample(
+                <<<'CODE_SAMPLE'
+{% set themeclass = item.title|replace(' ', '-') %}
+CODE_SAMPLE
+                ,
+                <<<'CODE_SAMPLE'
+{% set themeclass = item.title|replace({' ': '-'}) %}
+CODE_SAMPLE
+            ),
+        ]);
+    }
+
+    public function transform(string $twigContent): string
+    {
+        $twigContent = preg_replace('/replace\([\'"]{1}(.*)[\'"]{1},\s*[\'"]{1}(.*)[\'"]{1}\)/i', 'replace({\'$1\': \'$2\'})', $twigContent);
```

Comment:
I'm not sure how I feel about this being handled by a complex regex statement.

Is there support in Rector to parse twig?

It could be helpful to parse these twig deprecations, but the power of Rector seems to be the ability to avoid regex in favor of these parsing libraries that are more reliable?